### DISCUSSION TITLE
Questions on the Enhanced Protocol

### DISCUSSION THREAD

**Discussion by guido4096** [john30/ebusd#883] [Q&A]
The documentation [here](https://github.com/john30/ebusd/blob/master/docs/enhanced_proto.md) leaves some questions open. So far I have two:

**Question 1**
When the command START with SYN is send, ongoing arbitration has to be cancelled.
Does ebusd expect an answer FAILED in this case?

**Question 2**
There is a sentence "Note that this message shall not be sent when the byte received was part of an arbitration request initiated by ebusd." for the RECEIVED command
What is meant with "byte recevied was part of an arbitration request"? For example:
- After a START request, there is still data on the bus. Should this data be sent to ebusd?
- Does ebusd expect to receive the starting SYN ? Or is this considered part of the arbitration request?

**Comment by guido4096** [comment]
After testing, I believe the following is true on question 2:
- After receiving the START command, 
- The adapter should keep sending data to ebusd until the bus is ready to start arbitration

Once the adapter starts the arbitration
- The SYN symbol that starts the arbitration should also be send to ebusd
- At the end of arbitration, either a STARTED or a FAILED message has to be send
- Nothing else should be send in between the SYN and the STARTED/FAILED.

Will appreciate confirmation this is correct.

**Reply by tbertels** [reply]
> **Question 1** When the command START with SYN is send, ongoing arbitration has to be cancelled. Does ebusd expect an answer FAILED in this case?

Based on https://github.com/john30/ebusd/blob/ca6767dece2bee01ad8293f386ef09435f98f614/src/lib/ebus/device.cpp#L411-L424
I'd say only if it failed canceling the arbitration.

**Comment by guido4096** [comment]
thanks!
